### ISSUE TITLE
Async test functions are not checked

### ISSUE THREAD

**Issue by svazquezco** [jamescooke/flake8-aaa#244]
## Current behaviour
`flake8-aaa` does not appear to check tests declared with `async def`.

Async test functions are represented by Python's AST as `ast.AsyncFunctionDef`, but the current test discovery only visits `ast.FunctionDef`, so `async def test_*` functions are skipped.

Minimal reproduction:

```python
async def test_async_missing_act():
    setup = 1

    await thing()

    assert setup == 1


def test_sync_missing_act():
    setup = 1

    thing()

    assert setup == 1
```

Running:

```bash
flake8 test_async_aaa.py --select AAA
```
Only reports the synchronous test:

```text
test_async_aaa.py:8:1: AAA01 no Act block found in test
```

The async test is not reported.

## Expected behaviour

Both test functions should be checked.

The async test should report the same AAA violation as the synchronous test, because neither has a recognized Act block.

## Debugging info

Output of `python --version`:

```
Python 3.12.12
```

Output of `flake8 --version`:

```
7.3.0 (Flake8-pyproject: 1.2.4, flake8-aaa: 0.17.2, mccabe: 0.7.0,
pycodestyle: 2.14.0, pyflakes: 3.4.0, wemake-python-styleguide: 1.6.2) CPython
3.12.12 on Darwin
```

## Additional context
I checked the current master branch and TestFuncLister only defines visit_FunctionDef:

https://github.com/jamescooke/flake8-aaa/blob/master/src/flake8_aaa/helpers.py

A likely fix would be to add equivalent handling for visit_AsyncFunctionDef and update the affected type hints to accept both ast.FunctionDef and ast.AsyncFunctionDef.

**Comment by svazquezco** [comment]
@jamescooke I've already created the PR to fix the bug 😄 -> https://github.com/jamescooke/flake8-aaa/pull/245

**Comment by jamescooke** [comment]
Thanks for the feature request ❤️ 

And Yes - I agree it would be good if async test functions could be linted 👍🏻 . The reproduction example was helpful, but the PR in #245 is incomplete so I'm going to close it. Before this can be implemented there is maintenance work to do to drop Python 3.8 and 3.9, plus add the latest version, etc... all needed so that this feature can be added safely.

My plan is to migrate this repo off GH, so will have a look at this again when that is completed.
